File: src/main/kotlin/dev/jefy/connectpro/management/appliacaion/query/CategoryQuery.java
package dev.jefy.connectpro.management.appliacaion.query;

import org.jspecify.annotations.NullMarked;

import java.util.List;

import dev.jefy.connectpro.management.appliacaion.dtos.CategoryResponse;
import dev.jefy.connectpro.management.domain.vo.CategoryId;

/**
 * @author Jôph Yamba
 */
@NullMarked
public interface CategoryQuery {
    CategoryResponse get(CategoryId categoryId);
    List<CategoryResponse> getAll();
}

File: src/main/kotlin/dev/jefy/connectpro/management/appliacaion/query/CategoryQuery.kt
package dev.jefy.connectpro.management.appliacaion.query

import dev.jefy.connectpro.management.appliacaion.dtos.CategoryResponse
import dev.jefy.connectpro.management.domain.vo.CategoryId
import org.jspecify.annotations.NullMarked

/**
 * @author Jôph Yamba
 */

interface CategoryQuery {
    fun get(categoryId: CategoryId): CategoryResponse
    fun getAll(): List<CategoryResponse>
}
